safekeeper/src/send_wal.rs

```diff
@@ -191,100 +194,149 @@ impl ReplicationConn {
                 }
             })?;
 
-        let mut wal_seg_size: usize;
-        loop {
-            wal_seg_size = spg.timeline.get().get_state().1.server.wal_seg_size as usize;
-            if wal_seg_size == 0 {
-                error!("Cannot start replication before connecting to wal_proposer");
-                sleep(Duration::from_secs(1));
-            } else {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
+
+        runtime.block_on(async move {
+            let persisted_state: SafeKeeperState;
+            loop {
+                let (_, state) = spg.timeline.get().get_state();
+                if state.server.wal_seg_size == 0 || state.timeline_start_lsn == Lsn(0) {
+                    error!("Cannot start replication before connecting to wal_proposer");
+                    sleep(Duration::from_secs(1)).await;
```

Comment:
I wonder should we sleep and wait here until walproposer has connected, or it's better to drop the connection right away?

On the other side we likely have walreceiver who will retry on closed connection, so it might be a better way.

Reply:
+1 on dropping right away